src/clj/witan/send/charts.clj

```diff
@@ -107,4 +107,74 @@
+        (= (:calendar-year row) year)))
+
+(defn foo [year transitions-data]
+  (->> transitions-data
+       (filter #(is-send-and-is-year? % year))
+       (map :academic-year-1)
+       count))
+
+(defn population-line-plot
+  [historic-data future-projections-data]
+  (let [historic-years (vec (distinct (map :calendar-year historic-data)))
+        projected-years (vec (#(range % (+ (count future-projections-data) %))
+                         (inc (last historic-years))))
+        historic-pops (map #(get-total-historical-send-pop % historic-data) historic-years)
+        years (into [] (concat historic-years projected-years))
+        means (into [] (concat historic-pops (map int (map :mean future-projections-data))))
+        low-cis (into [] (concat (repeat (count historic-years) :NA) (map :low-ci future-projections-data)))
+        high-cis (into [] (concat (repeat (count historic-years) :NA) (map :high-ci future-projections-data)))
+        gg4clj-data-frame {:year years :mean means :min low-cis :max high-cis}
+        min-year (first historic-years)
+        max-year (last projected-years)
+        max-max (apply max (map :max future-projections-data))]
+    (gg4clj/render [[:require "ggforce"]
+                    [:<- :df
+                     (gg4clj/data-frame gg4clj-data-frame)]
+                     (gg4clj/r+ [:ggplot {:data :df} [:aes {:x :year :y :mean}]]
+                                [:ggtitle "SEND Population Projection"]
+                                [:geom_line [:aes {:colour "mean" :linetype "mean"}]]
+                                [:geom_point {:colour "darkcyan"}]
+                                [:geom_line {:data :df} [:aes {:x :year :y :max :colour "conf" :linetype "conf"}]]
+                                [:geom_line {:data :df} [:aes {:x :year :y :min :colour "conf" :linetype "conf"}]]
+                                [:scale_y_continuous {:name "Total SEND population" :limits [:c 0 max-max]}]
+                                [:scale_x_continuous {:name "Year" :breaks [:seq min-year max-year {:by 1}] :limits [:c min-year max-year]}]
+                                [:theme_bw]
+                                [:scale_linetype_manual {:name "" :values [:c {:mean "solid" :conf "dashed"}] :labels [:c {:mean "Mean" :conf "95% Confidence"}]}]
+                                [:scale_colour_manual {:name "" :values [:c {:mean "darkcyan" :conf "grey38"}] :labels [:c {:mean "Mean" :conf "95% Confidence"}]}])]))
+    (move-file "Rplots.pdf" "target/Total_Population.pdf"))
+ 
+(defn send-cost-plot
+  [cost-data years]
+  (let [proj-years (vec (#(range % (+ (count cost-data) %)) (inc (last years))))
+        low-ci (vec  (map #(/ % 1000000) (map :low-ci cost-data)))
+        q1 (vec (map #(/ % 1000000) (map :q1 cost-data)))
+        median  (vec (map #(/ % 1000000) (map :median cost-data)))
+        q3 (vec (map #(/ % 1000000) (map :q3 cost-data)))
+        high-ci (vec (map #(/ % 1000000) (map :high-ci cost-data)))
```

Comment:
Sorry I haven't looked elsewhere, but why `(/ % 1000000)`?

Reply:
This gives the costs in millions, as with the Excel chart it copies,  and avoids an enormous y-axis.